skins/laika/src/components/pages/donation_form/Email.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+    <div>
+        <h1 class="title is-size-1">{{ $t( 'donation_section_email_title' ) }}</h1>
+        <div>
+            <label for="email" class="subtitle has-margin-top-18">{{ $t( 'email_label' ) }}</label>
+            <b-input type="text"
+                id="email"
+                name="email"
+                :placeholder="$t( 'email_label' )"
+                v-model="emailValue"
+                @blur="validateEmail">
+            </b-input>
+            <span v-if="emailHasError" class="help is-danger">{{ $t( 'form_email_error' ) }}</span>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { NS_ADDRESS } from '@/store/namespaces';
+import { action } from '@/store/util';
+import { setEmail, setNewsletterOptIn } from '@/store/address/actionTypes';
+
+export default Vue.extend( {
+	name: 'Email',
+	data: function () {
+		return {
+			emailValue: '',
+			emailPattern: /^[^@]+@.+$/,
+			emailHasError: false,
+		};
+	},
+	methods: {
+		setEmail: function () {
+			this.$store.dispatch( action( NS_ADDRESS, setEmail ), this.$data.emailValue );
+		},
+		validateEmail: function () {
+			if ( this.emailIsValid() ) {
+				this.emailHasError = false;
+				this.setEmail();
+			} else { this.emailHasError = true; }
```

Comment:
You could avoid the `else` here:

```
this.emailHasError = !this.emailIsValid();
if ( !this.emailHasError ) {
     this.setEmail();
}
```

I don't think it's a super big deal but the method also validates and sets the email, maybe it could also just consolidate the set and validate function into one because I guess this would be more accurate?!

```
setEmail: function () {
    this.emailHasError = !this.emailIsValid();
    if ( !this.emailHasError ) {
        this.$store.dispatch( action( NS_ADDRESS, setEmail ), this.$data.emailValue );
    }
```

Reply:
I'd like to keep `setEmail` clean a.k.a doing only one thing.
That's why I made `emailIsValid` and `validateEmail` to glue the logic together.

Reply:
Makes sense!